include/itkTileMontage.hxx

```diff
@@ -472,7 +449,213 @@ void
+    if ( this->GetDebug() )
+      {
+      std::cout << "\nstdDev0:\n" << stdDev0;
+      }
+
+    std::vector< TCoordinate > outlierScore( nReg, 0.0 ); // sum of squares
+    for ( SizeValueType i = 0; i < nReg; i++ )
+      {
+      for ( unsigned d = 0; d < ImageDimension; d++ )
+        {
+        TCoordinate trOverDev = translations( i, d ) / stdDev0( d );
+        outlierScore[i] += std::abs( trOverDev );
+        }
+      if ( outlierScore[i] > m_RelativeThreshold ) // more than this many standard deviations
+        {
+        outlierScore[i] -= m_RelativeThreshold;
+        }
+      else
+        {
+        outlierScore[i] = 0.0; // not an outlier
+        }
+      }
+
+    TCoordinate maxCost = 0;
+    SizeValueType maxIndex;
+    if ( this->GetDebug() )
+      {
+      std::cout << "\nresiduals:\n";
+      }
+
+    for ( SizeValueType i = 0; i < nReg; i++ )
+      {
+      TCoordinate residual = 0;
+      if ( this->GetDebug() )
+        {
+        std::cout << 'E' << i << ':';
+        }
+      for ( unsigned d = 0; d < ImageDimension; d++ )
+        {
+        if ( this->GetDebug() )
+          {
+          std::cout << ' ' << std::setw( 8 ) << residuals( i, d );
+          }
+        residual += residuals( i, d ) * residuals( i, d );
+        }
+      residual = std::sqrt( residual ); // MSE -> RMSE
+      TCoordinate cost = residual + std::sqrt( residual * outlierScore[i] ) + outlierScore[i];
+      if ( this->GetDebug() )
+        {
+        std::cout << " :" << std::setw( 6 ) << outlierScore[i];
+        std::cout << " =" << std::setw( 8 ) << cost;
+        std::cout << std::endl;
+        }
+      if ( cost > maxCost )
+        {
+        maxCost = cost;
+        maxIndex = i;
+        }
+      }
+    if ( this->GetDebug() )
+      {
+      std::cout << std::endl;
+      }
+
+    static float const sqrtDim = std::sqrt( ImageDimension );
+    if ( maxCost < m_AbsoluteThreshold * sqrtDim )
+      {
+      outlierExists = false;
+      }
+    else // eliminate the problematic equation
+      {
+      SizeValueType candidateIndex = equationToCandidate[maxIndex];
+      std::cout << "Outlier detected. Equation " << maxIndex << ", Registration " << candidateIndex << ", T: ";
+      if ( !m_TransformCandidates[candidateIndex].empty() )
+        {
+        std::cout << m_TransformCandidates[candidateIndex][0];
+        m_TransformCandidates[candidateIndex].erase( m_TransformCandidates[candidateIndex].begin() );
+        }
+      else
+        {
+        std::cout << "zeroes";
+        }
+
+      if ( !m_TransformCandidates[candidateIndex].empty() )
+        {
+        // get a new equation from m_TransformCandidates
+        const TranslationOffset& candidateOffset = m_TransformCandidates[candidateIndex][0];
+        for ( unsigned d = 0; d < ImageDimension; d++ )
+          {
+          translations( maxIndex, d ) = candidateOffset[d];
+          }
+        std::cout << "  Replaced by T: " << candidateOffset;
```

Comment:
Remove debug code

Reply:
This is more of a legitimate info message, than debug code.